bus/rabbitmq/queue.js

```diff
@@ -145,19 +146,21 @@ Queue.prototype.unlisten = function unlisten () {
   return em;
 };
 
-Queue.prototype.send = function send (event, options) {
+Queue.prototype.send = function send (event, options, cb) {
   options = options || {};
   var self = this;
 
   if ( ! this.initialized) {
-    return this.on('ready', send.bind(this, event, options));
+    return this.on('ready', send.bind(this, event, options, cb));
   }
 
   options.contentType = options.contentType || this.contentType;
   options.persistent = Boolean(options.ack || options.acknowledge || options.persistent || self.ack);
 
-  this.sendChannel.sendToQueue(this.routingKey || this.queueName, new Buffer(options.formatter.serialize(event)), options);
-  
+  var channel = cb ? this.confirmChannel : this.sendChannel;
```

Comment:
Can you check this.confirmChannel && cb? We don't want to try to use the confirmChannel just because they tried using a callback on a non-confirm queue.